cmd/api/src/queries/graph.go

```diff
@@ -773,67 +772,72 @@ func RunListQuery(ctx context.Context, db graph.Database, delegate any, node *gr
 	return result, nil
 }
 
-func (s *GraphQuery) runListEntityQuery(ctx context.Context, node *graph.Node, params EntityQueryParameters, cacheEnabled bool) (any, error) {
+func (s *GraphQuery) runMaybeCachedEntityQuery(ctx context.Context, node *graph.Node, params EntityQueryParameters, cacheEnabled bool) (graph.NodeSet, error) {
 	var (
-		cacheKey   = fmt.Sprintf("ad-entity-query_%s_%s_%d", params.QueryName, params.ObjectID, params.RequestedType)
 		queryStart = time.Now()
-		skip       = params.Skip
-		limit      = params.Limit
-		mustFetch  = !cacheEnabled
+		cacheKey   = fmt.Sprintf("ad-entity-query_%s_%s_%d", params.QueryName, params.ObjectID, params.RequestedType)
+
+		foundResultInCache = false
 
 		result graph.NodeSet
 	)
 
 	if cacheEnabled {
-		if hasResult, err := s.Cache.Get(cacheKey, &result); err != nil {
+		var err error
+		if foundResultInCache, err = s.Cache.Get(cacheKey, &result); err != nil {
 			return nil, fmt.Errorf("error getting cache entry for %s: %w", cacheKey, err)
-		} else {
-			mustFetch = !hasResult
 		}
 	}
 
-	if mustFetch {
+	if !cacheEnabled || !foundResultInCache {
 		// Fetch the entire result for caching purposes
-		if fetchedResult, err := RunListQuery(ctx, s.Graph, params.ListDelegate, node, 0, 0); err != nil {
+		if fetchedResult, err := runEntityQuery(ctx, s.Graph, params.ListDelegate, node, 0, 0); err != nil {
 			return nil, err
 		} else {
 			result = fetchedResult
 		}
 	}
 
-	// Return early if this is just a count request
-	if params.RequestedType == model.DataTypeCount {
```

Comment:
These requests were never cached previously because they always broke out early. By pulling out this to it's own function, we are able to now cache the DataTypeCount responses as well.